ui/apps/platform/src/Components/PatternFly/CheckboxSelect.tsx

```diff
@@ -101,20 +134,13 @@ function CheckboxSelect({
         </MenuToggle>
     );
 
-    // Automatically inject hasCheckbox and isSelected props
-    const enhancedChildren = React.Children.map(children, (child) => {
-        if (React.isValidElement(child) && child.type === SelectOption) {
-            const { value } = child.props;
-            if (value != null) {
-                return React.cloneElement(child, {
-                    hasCheckbox: true,
-                    isSelected: selections.includes(value as string),
-                    ...child.props, // Allow explicit overrides if needed
-                });
-            }
-        }
-        return child;
-    });
+    // Convert selections to Set for O(1) lookup performance
+    const selectionsSet = useMemo(() => new Set(selections), [selections]);
```

Comment:
The NamespaceSelector and DeploymentSelector can potentially have 100+ select options. Having an O(1) lookup should help significantly with performance.

We also have our enhanceSelectOptions function that needs to modify the children to add props for the Checkbox Select options. This should be helpful for that.